web/components/new-contract-panel.tsx

```diff
@@ -220,13 +223,22 @@ export function NewContractPanel(props: {
 
   const [hideOptions, setHideOptions] = useState(true)
 
-  const [permittedGroups, setPermittedGroups] = useState<Group[]>([])
+  const [adminGroups, setAdminGroups] = useState<Group[]>([])
+  const [publicGroups, setPublicGroups] = useState<Group[]>([])
   useEffect(() => {
-    getGroupsWhereUserIsMember(creator.id).then((g) =>
-      setPermittedGroups(g.map((gp: { group_data: any }) => gp.group_data))
-    )
+    getNonPublicGroupsWhereUserHasRole(creator.id)
+      .then((g) =>
+        setAdminGroups(g.map((gp: { group_data: any }) => gp.group_data))
+      )
+      .catch((e) => console.log(e))
   }, [creator.id])
 
+  useEffect(() => {
+    getPublicGroups()
+      .then((g) => setPublicGroups(g.map((gp: { data: any }) => gp.data)))
```

Comment:
If you do this transformation everywhere, shouldn't the `getPublicGroups` function do it for you?

Reply:
Same for `getNonPublicGroupsWhereUserHasRole`